tools/myst/config.c

```diff
@@ -423,6 +423,15 @@ static json_result_t _json_read_callback(
                 else
                     CONFIG_RAISE(JSON_TYPE_MISMATCH);
             }
+            else if (json_match(parser, "HostUDS") == JSON_OK)
+            {
+                if (type == JSON_TYPE_BOOLEAN)
+                    parsed_data->host_uds = un->boolean;
+                else if (type == JSON_TYPE_INTEGER)
+                    parsed_data->host_uds = (un->integer == 0) ? false : true;
```

Comment:
If the type of ``parsed_data->host_uds`` is ``bool``, then the following will be fine.

```
parsed_data->host_uds = (un->integer == 0);
```
The compiler automatically injects code to set ``bool`` to ``0`` or ``1``. The following program prints ``1``.

```
#include <stdio.h>
#include <stdbool.h>

int main()
{
    printf("%u\n", (bool)99);
    return 0;
}
```